src/libraries/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/DelimitedListTraceListener.cs

```diff
@@ -48,9 +49,13 @@ public string Delimiter
                 {
                     if (!_initializedDelim)
                     {
-                        if (Attributes.ContainsKey("delimiter"))
+                        if (Attributes.ContainsKey(DelimiterKey))
                         {
-                            _delimiter = Attributes["delimiter"];
+                            string? delimiter = Attributes[DelimiterKey];
+                            if (!string.IsNullOrEmpty(delimiter))
```

Comment:
> This looks like a (corner-case) change of behavior.

Yes and no.  This code was actually only brought back for .NET 5 in https://github.com/dotnet/runtime/pull/635; before that, Attributes wasn't consulted in .NET Core at all.  So this PR's change won't impact previous code running on .NET Core, though this will be a corner-case difference from .NET Framework.  However, to me this is a bug fix more than anything else: the setter of this property explicitly throws for null, which means null was never intended to be a valid Delimiter, yet there's this unintentional backdoor where setting null as the "delimiter" key into the attributes collection could erroneously introduce null.  Seemed better to just fix it.